### PR TITLE
osc/ucx:  flush all the outstanding transfers before cleaning up the thread local context

### PR DESCRIPTION
**Pull request by MamziB**

### CODE REVIEW

**Comment by awlauria** [comment]
bot:aws:retest

**Review by janjust — APPROVED** [review]

File: opal/mca/common/ucx/common_ucx_wpool.c [hunk 1]
```diff
@@ -562,6 +562,16 @@ static void _tlocal_ctx_rec_cleanup(_ctx_record_t *ctx_rec)
     opal_common_ucx_winfo_t *winfo = ctx_rec->winfo;
     opal_common_ucx_wpool_t *wpool = ctx_rec->gctx->wpool;
 
+    opal_mutex_lock(&winfo->mutex);
+    int rc = opal_common_ucx_winfo_flush(winfo, 0, OPAL_COMMON_UCX_FLUSH_B, OPAL_COMMON_UCX_SCOPE_WORKER, NULL);
+    winfo->global_inflight_ops = 0;
+    memset(winfo->inflight_ops, 0, winfo->comm_size * sizeof(short));
+    opal_mutex_unlock(&winfo->mutex);
+    if (rc != OPAL_SUCCESS) {
```
Comment (gpaulsen) [inline opal/mca/common/ucx/common_ucx_wpool.c:570]: It'd be better to move the check for failure inside of the mutex (but of course remember to unloc before returning).

**Review by gpaulsen — CHANGES_REQUESTED** [review]

File: opal/mca/common/ucx/common_ucx_wpool.c (hunk 1 above)
Comment (awlauria) [reply opal/mca/common/ucx/common_ucx_wpool.c:570]: `rc` is a local variable, so it is immune to being changed by another thread for what it is worth.

Comment (janjust) [reply opal/mca/common/ucx/common_ucx_wpool.c:570]: yea I was going to ask - why move it inside the mutex?

**Review by awlauria — APPROVED** [review]

**Review by gpaulsen — APPROVED** [review]
I'm not too passionate about moving the check for error inside the mutex.

**Pull request merged**